Content feat

wenodev opened the pull request:
Content와 Probelm의 연관관계를 생성하던 중 궁금한점이 생겼습니다.
1. Content와 Problem를 ManyToOne으로 하게되면 Problem의 요청 및 응답 DTO에 Content DTO가 포함되면 되는걸까요?
2. 그렇게 된다면 상황에 따라 다르겠지만 Content의 Controller와 Service는 생성할 필요가 없던게 맞을까요?
3. ProblemSerivce에서 ContentRepository와 ContetService 중 어떤걸 주입 받아야 할까요?

wenodev committed:
feat: getAllContents 생성하라

wenodev committed:
test : getAllContents 성공하라

wenodev committed:
feat: ContentRequestDto 생성하라

wenodev committed:
feat: saveContent 생성하라

wenodev committed:
feat: updateContent 생성하라

wenodev committed:
feat: Entity의 updateContent 생성하라

wenodev committed:
feat: deleteContent 생성하라

wenodev committed:
test: deleteContent 성공하라

wenodev committed:
feat: contentController를 위한 fixture를 생성하라

wenodev committed:
test: getContent 실패하라

wenodev committed:
feat: getContent 생성하라

wenodev committed:
test: getAllContent 실패하라

wenodev committed:
feat: getAllContent 생성하라

wenodev committed:
test: saveContent 실패하라

wenodev committed:
feat: saveContent 생성하라

wenodev committed:
test: updateContent 실패하라

wenodev committed:
feat: updateContent 생성하라

wenodev committed:
test: deleteContent 실패하라

wenodev committed:
feat: deleteContent 생성하라

wenodev committed:
fix: delete의 응답코드를 204로 변경하라

wenodev committed:
feat: problem 칼럼 생성하라

ManyToOne 단방향 관계를 생성합니다.

wenodev committed:
fix: ProblemResponseDto에 ContentResponseDto를 추가하라

wenodev committed:
feat: ProblemRequestDto에 ContentRequestDto를 추가하라

ContentRequestDto의 변수명은 contents로 했으며 프론트엔드에서 보낼때 네이밍을 편히라게 하기 위함입니다.

wenodev committed:
feat: 생성자에 Problem 추가하라

wenodev committed:
fix: ContentRequestDto를 List로 선언하라

wenodev committed:
test: saveProblem 실패하라

saveProblem에 Content 저장하는 테스트를 진행합니다.

johngrib commented:
> Content와 Probelm의 연관관계를 생성하던 중 궁금한점이 생겼습니다.
> 
> 1. Content와 Problem를 ManyToOne으로 하게되면 Problem의 요청 및 응답 DTO에 Content DTO가 포함되면 되는걸까요?
> 2. 그렇게 된다면 상황에 따라 다르겠지만 Content의 Controller와 Service는 생성할 필요가 없던게 맞을까요?
> 3. ProblemSerivce에서 ContentRepository와 ContetService 중 어떤걸 주입 받아야 할까요?

오 설계에 대한 고민이 시작됐군요. 이건 정답이 없습니다. 한 번에 답이 나오지도 않습니다. 좀 괴상한 방법처럼 들릴 수 있는데요, 3번 정도 프로젝트 전체를 다시 만들어 보면 대강 이렇게 해야겠다는 게 잡히긴 합니다.

johngrib reviewed (commented):
주요 클래스에 JavaDoc 주석을 작성해주시면 다른 사람이 조금 더 빨리 파악할 수 있습니다.

wenodev commented:
> > Content와 Probelm의 연관관계를 생성하던 중 궁금한점이 생겼습니다.
> > 
> > 1. Content와 Problem를 ManyToOne으로 하게되면 Problem의 요청 및 응답 DTO에 Content DTO가 포함되면 되는걸까요?
> > 2. 그렇게 된다면 상황에 따라 다르겠지만 Content의 Controller와 Service는 생성할 필요가 없던게 맞을까요?
> > 3. ProblemSerivce에서 ContentRepository와 ContetService 중 어떤걸 주입 받아야 할까요?
> 
> 오 설계에 대한 고민이 시작됐군요. 이건 정답이 없습니다. 한 번에 답이 나오지도 않습니다. 좀 괴상한 방법처럼 들릴 수 있는데요, 3번 정도 프로젝트 전체를 다시 만들어 보면 대강 이렇게 해야겠다는 게 잡히긴 합니다.

그렇군요. 계속해서 수련하겠습니다!

wenodev commented:
> 주요 클래스에 JavaDoc 주석을 작성해주시면 다른 사람이 조금 더 빨리 파악할 수 있습니다.

넵!

wenodev committed:
test: saveProblem 실패하라

wenodev committed:
feat: toEntityList 생성하라

Content로 변환할때 List형태로 변환할 상황이 많을것으로 예상되 추가합니다.

wenodev committed:
fix: ProblemResponseDto에 contentResponseDto를 리턴하도록 하라

wenodev committed:
fix: saveProblem을 리턴할때 ContentResponseDto를 함께 리턴하도록 하라

johngrib commented on app/src/test/java/com/weno/problem/ProblemServiceTest.java:
```diff
@@ -85,7 +84,8 @@ void testSaveProblem(){
         ProblemResponseDto problemResponse = problemService.saveProblem(problemRequest);
 
-        assertThat(problemResponse.getTitle()).isEqualTo("dummy-test-title-2");
+        assertThat(problemResponse.getTitle()).isEqualTo("dummy-test-title-new");
```
`isEqualTo`에 넘겨주는 값을 이렇게 작성하면 테스트가 오타에 취약해집니다.

wenodev replied:
> `isEqualTo`에 넘겨주는 값을 이렇게 작성하면 테스트가 오타에 취약해집니다.

final로 선언하는 방법을 고민해보겠습니다.

wenodev committed:
test: 모든 Problems를 불러오는 로직을 실패하라

wenodev committed:
test: 모든 Problems를 불러오는 테스트 코드를 성공하라

Problems를 불러올때 Contents를 불러오지 못했다.
ProblemResponseDto에 ContentResponseDto를 추가하여 같이 리턴하도록 수정했다.

wenodev committed:
fix: 모든 Problems 리턴할때 DTO로 변경하여 리턴하도록 수저하라

wenodev committed:
fix: Content 저장할때 Problem정보를 같이 저장하도록 하라

wenodev committed:
fix: Entity에서 데이터 수정할때 Dto와 관련없도록 수정하라

wenodev committed:
chore: h2-console에 접속하도록하고 메모리 DB를 사요하도록 변경하라

wenodev committed:
fix: Contetnt update 로직 추가하라

wenodev committed:
test: Probelm update 테스트코드 성공하라

wenodev committed:
fix: 디렉토리 이름 변경하라
